osiris/lib/kafka/consumer.go

```diff
@@ -15,23 +16,25 @@ import (
 var KafkaConsumerCliOptions = libcli.CommonCliOptions(libcli.Kafka)
 
 type KafkaConsumerOptions struct {
-	Brokers []string
-	Group   string
-	Version string
-	Topics  []string
+	Brokers             []string
+	Group               string
+	Version             string
+	Topics              []string
+	ConnectionTolerance libcli.ConnectionTolerance
 }
 
 func (config KafkaConsumerOptions) ParseCli(ctx *cli.Context) KafkaConsumerOptions {
 	return KafkaConsumerOptions{
-		Brokers: strings.Split(ctx.String("kafka-brokers"), ","),
-		Group:   ctx.String("kafka-group"),
-		Version: ctx.String("kafka-version"),
-		Topics:  strings.Split(ctx.String("kafka-topics"), ","),
+		Brokers:             strings.Split(ctx.String("kafka-brokers"), ","),
+		Group:               ctx.String("kafka-group"),
+		Version:             ctx.String("kafka-version"),
+		Topics:              strings.Split(ctx.String("kafka-topics"), ","),
+		ConnectionTolerance: libcli.ConnectionTolerance{}.ParseCli(ctx),
 	}
 }
 
 type KafkaConsumer struct {
-	ready    chan bool
+	ready    chan error
```

Comment:
Maybe rename ready to status or something similar, because it's not true or false anymore, but hold more information?

Reply:
hmmm, well